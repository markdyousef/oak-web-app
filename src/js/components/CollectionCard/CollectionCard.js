// @flow
import React, { PropTypes } from 'react';
import styled from 'styled-components';
import colors from '../../styles/colors';
import placeholder from '../../../img/collections-placeholder.svg';

const Container = styled.div`
    width: 320px;
    background: ${colors.white};
    padding: 30px;
    border: 1px solid ${colors.lightGrey};
    border-radius: 3px;
    height: 200px;
    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;
    &:hover {
        background: #fafafa;
    }
    &:active {
        background: #f7f7f7;
    }
`;

const Name = styled.h3`
    font-size: 24px;
    font-weight: normal;
    line-height: 1.48em;
    margin-bottom: 12px;
    color: ${colors.black};
    margin: 16px 0 12px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 224px;
`;

const Description = styled.p`
    font-size: 14px;
    color: ${colors.black};
    font-weight: 300;
    line-height: 1.54em;
    text-align: center;
    overflow: hidden;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    max-height: 3.2em;
`;

const Creator = styled.img`
    position: absolute;
    bottom: 16px;
    right: 16px;
    height: 24px;
    width: 24px;
    border-radius: 99em;
`;

const CollectionCard = ({ name, description, picture }:Object) => {
    console.log(picture);
    return (
        <Container>
            <Name>{name}</Name>
            <Description>{description}</Description>
            {picture && <Creator src={picture} alt="creator" />}
        </Container>
    );
};

CollectionCard.propTypes = {
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    picture: PropTypes.string
};

CollectionCard.defaultProps = {
    description: null,
    picture: null
};

export default CollectionCard;
