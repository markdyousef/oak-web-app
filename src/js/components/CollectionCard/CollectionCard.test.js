import React from 'react';
import renderer from 'react-test-renderer';
import CollectionCard from './CollectionCard';

it('renders correctly', () => {
    const props = {
        name: 'collection'
    };
    const component = renderer.create(
        <CollectionCard {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
