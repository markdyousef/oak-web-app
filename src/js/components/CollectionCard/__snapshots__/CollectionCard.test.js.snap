exports[`test renders correctly 1`] = `
<div
  className="bUuDYD">
  <h3
    className="ihqnMT">
    collection
  </h3>
  <p
    className="kqXysH" />
</div>
`;
